unit-tests/internal/internal-tests-usb.cpp

```diff
@@ -0,0 +1,270 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2015 Intel Corporation. All Rights Reserved.
+
+#include "catch/catch.hpp"
+#include "usb/usb-enumerator.h"
+#include "usb/usb-device.h"
+#include "hw-monitor.h"
+#include "librealsense2/h/rs_option.h"
+#include <map>
+
+using namespace librealsense;
+using namespace librealsense::platform;
+
+static std::map< std::string, int> uvc_req_code = {
+    {"UVC_RC_UNDEFINED", 0x00},
+    {"UVC_SET_CUR", 0x01},
+    {"UVC_GET_CUR", 0x81},
+    {"UVC_GET_MIN", 0x82},
+    {"UVC_GET_MAX", 0x83},
+    {"UVC_GET_RES", 0x84},
+    {"UVC_GET_LEN", 0x85},
+    {"UVC_GET_INFO", 0x86},
+    {"UVC_GET_DEF", 0x87},
+    {"UVC_REQ_TYPE_GET", 0xa1},
+    {"UVC_REQ_TYPE_SET", 0x21}
+};
+
+std::string req_to_string(int req)
+{
+    auto res = std::find_if(std::begin(uvc_req_code), std::end(uvc_req_code), [&](const std::pair<std::string, int> &p)
+    {
+        return p.second == req;
+    });
+    if (res != std::end(uvc_req_code))
+        return res->first;
+    return "";
+}
+
+static std::map< std::string, int> pu_controls = {
+    {"PU_CONTROL_UNDEFINED", 0x00},
+    {"PU_BACKLIGHT_COMPENSATION_CONTROL", 0x01},
+    {"PU_BRIGHTNESS_CONTROL", 0x02},
+    {"PU_CONTRAST_CONTROL", 0x03 },
+    {"PU_GAIN_CONTROL", 0x04 },
+    {"PU_POWER_LINE_FREQUENCY_CONTROL", 0x05 },
+    {"PU_HUE_CONTROL", 0x06 },
+    {"PU_SATURATION_CONTROL", 0x07 },
+    {"PU_SHARPNESS_CONTROL", 0x08 },
+    {"PU_GAMMA_CONTROL", 0x09 },
+    {"PU_WHITE_BALANCE_TEMPERATURE_CONTROL", 0x0A },
+    {"PU_WHITE_BALANCE_TEMPERATURE_AUTO_CONTROL", 0x0B },
+    {"PU_WHITE_BALANCE_COMPONENT_CONTROL", 0x0C },
+    {"PU_WHITE_BALANCE_COMPONENT_AUTO_CONTROL", 0x0D },
+    {"PU_DIGITAL_MULTIPLIER_CONTROL", 0x0E },
+    {"PU_DIGITAL_MULTIPLIER_LIMIT_CONTROL", 0x0F },
+    {"PU_HUE_AUTO_CONTROL", 0x10 },
+    {"PU_ANALOG_VIDEO_STANDARD_CONTROL", 0x11 },
+    {"PU_ANALOG_LOCK_STATUS_CONTROL", 0x12 },
+    {"PU_CONTRAST_AUTO_CONTROL", 0x13 },
+
+};
+
+TEST_CASE("query_devices", "[live][usb]") 
+{
+    auto  devices = usb_enumerator::query_devices();
+
+    REQUIRE(devices.size());
+
+    switch (devices.size())
+    {
+        case 0: printf("\nno USB device found\n\n"); break;
```

Comment:
This case will not be invoked - terminated on previous ` REQUIRE(devices.size());`

Reply:
removed